### ISSUE TITLE
Recurring "connection failed" error when opening Jupyter notebook

### ISSUE THREAD

**Issue by oC-n** [alan-turing-institute/rse-course#189]
When I run `jupyter notebook` from the `rse-course` environment, it successfully opens the Home Page (index), but almost every time I open a notebook, I get a "Connection failed" error: "A connection to the notebook server could not be established. The notebook will continue trying to reconnect. Check your network connection or notebook server configuration."

My default browser is Firefox (latest version), running on M1 MacBook Air, MacOS 12.6.2.

The HTTP requests are giving a whole lot of 403 Forbidden errors: `403 GET /api/terminals?_=1673524153596 (127.0.0.1) 2.930000ms referer=http://localhost:8888/tree`

and then says the notebook isn't trusted:
`Notebook 01_00_introduction_to_python.ipynb is not trusted`

Here's the full HTTP record from the initial connection:
```
[W 11:50:08.773 NotebookApp] Forbidden
[W 11:50:08.774 NotebookApp] 403 GET /api/sessions?_=1673524153595 (127.0.0.1) 2.440000ms referer=http://localhost:8888/tree
[W 11:50:08.774 NotebookApp] Forbidden
[W 11:50:08.775 NotebookApp] 403 GET /api/terminals?_=1673524153596 (127.0.0.1) 2.930000ms referer=http://localhost:8888/tree
[W 11:50:12.554 NotebookApp] Notebook 01_00_introduction_to_python.ipynb is not trusted
/Users/deepthought/anaconda3/envs/rse-course/lib/python3.8/site-packages/jupyter_server/base/handlers.py:240: UserWarning: The Tornado web application does not have an 'identity_provider' defined in its settings. In future releases of jupyter_server, this will be a required key for all subclasses of `JupyterHandler`. For an example, see the jupyter_server source code for how to add an identity provider to the tornado settings: https://github.com/jupyter-server/jupyter_server/blob/aa8fd8b3faf37466eeb99689d5555314c5bf6640/jupyter_server/serverapp.py#L253
  warnings.warn(
[E 11:50:12.858 NotebookApp] Error unpacking user from cookie: Extra data: line 1 column 3 (char 2)
[W 11:50:12.858 NotebookApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:50:12.861 NotebookApp] 403 POST /nbdime/api/isgit (127.0.0.1) 3.720000ms referer=http://localhost:8888/notebooks/01_00_introduction_to_python.ipynb
[I 11:50:12.868 NotebookApp] Kernel started: 50485ac7-9037-4415-94b8-fea006f5cf77, name: python3
[W 11:50:12.884 NotebookApp] Forbidden
[W 11:50:12.885 NotebookApp] 403 GET /api/contents/01_00_introduction_to_python.ipynb/checkpoints?_=1673524212421 (127.0.0.1) 0.690000ms referer=http://localhost:8888/notebooks/01_00_introduction_to_python.ipynb
[W 11:50:12.911 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:12.914 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 3.640000ms referer=None
[W 11:50:14.407 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:14.408 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 2.670000ms referer=None
[W 11:50:16.643 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:16.645 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 4.400000ms referer=None
[W 11:50:20.664 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:20.666 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 2.960000ms referer=None
[W 11:50:28.778 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:28.779 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 2.880000ms referer=None
[W 11:50:44.849 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:50:44.852 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 3.530000ms referer=None
[W 11:51:16.978 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:51:16.979 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 2.800000ms referer=None
[W 11:52:21.989 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:52:21.990 NotebookApp] 403 GET /api/kernels/50485ac7-9037-4415-94b8-fea006f5cf77/channels?session_id=de11203641b04979bc612e62d10133bf (127.0.0.1) 2.300000ms referer=None
```

For a while it worked, after I set Firefox to turn off Enhanced Tracking Protection, but it reverted to failing. The issue persisted after clearing cookies and multiple restarts of the notebook server.

Likewise, opening in Chrome worked briefly, then instead of opening each requested notebook, it instead presented a "Token authentication is enabled" page, asking for the token. If I copy in the token from the localhost URL, it opens the page, but immediately displays the same "Connection failed" error as above, with a whole lot more "Couldn't authenticate WebSocket connection" responses.

Basically, I can't run the Jupyter Notebooks in a browser. Fortunately, all seems ok in opening the Notebooks in VS Code using the Jupyter Notebooks plugin.

**Comment by jemrobinson** [comment]
This looks like a `jupyter` configuration issue rather than an issue with the notebooks in this repository which are working for other people. Have you tried reporting this to https://jupyter.org/community ?
